src/wallet/db.cpp

```diff
@@ -349,13 +349,26 @@ void BerkeleyEnvironment::CheckpointLSN(const std::string& strFile)
 
 BerkeleyBatch::BerkeleyBatch(BerkeleyDatabase& database, const char* pszMode, bool fFlushOnCloseIn) : pdb(nullptr), activeTxn(nullptr)
 {
+    database.Open(pszMode);
     fReadOnly = (!strchr(pszMode, '+') && !strchr(pszMode, 'w'));
     fFlushOnClose = fFlushOnCloseIn;
     env = database.env.get();
-    if (database.IsDummy()) {
+    pdb = database.m_db.get();
+    strFile = database.strFile;
+    if (strchr(pszMode, 'c') != nullptr && !Exists(std::string("version"))) {
```

Comment:
3cdb7fd9d1e92f07c04c325c9b585ff736162198  nit: the `fCreate` helper variable makes this more readable. It's introduced in a later commit, but might as well do it here.

Reply:
Done